### ISSUE TITLE
Is there a way to use "dynamic function definiton macro" for test cases ?

### ISSUE THREAD

**Issue by ranserme** [ThrowTheSwitch/Ceedling#269]
Hi, 
My goal is to generate several test cases using the following kind of macro:

*test_file.c*:
#define MY_TEST_MACRO(name, param, expected_res) \\      
void test_MY_TEST_##name(void) \\
{\\
	uint8_t uRes = a_function_to_test(param);\\
        TEST_ASSERT(uRes == expected_res); \\ 
}

and then define my test cases like : 

MY_TEST_MACRO(case1, 28, 32)
MY_TEST_MACRO(case2, 27, 33)
MY_TEST_MACRO(case3, 12, 56)

*********************************************************************************************************

My test_file seems properly pre-processed as the wanted functions are properly declared in my **test_file.c** located in  the build/test/preprocess/**test_file.c**

void test_MY_TEST_case1(void){ /*test code*/}
void test_MY_TEST_case2(void){/*test code*/}
void test_MY_TEST_case3(void){/*test code*/}

But unfortunately none of the functions defined with the macro are present in the build/test/test_file_runner.c file => none of them are executed.

=> I'd guess the runner file is generated prior the preprocess/**test_file.c**  
=> Is there a way to force a new generation of the runner file from the pre-processed *test_file.c* ?

I'm looking forward to your feedback. 
Best Regards

**Comment by mvandervoord** [comment]
You are correct. The Test Runner Generator parses the test file itself, it doesn't do anything with preprocessed data, so it won't know anything about your expanded macros.

You can accomplish the same thing by creating the tests and just removing the name portion of your macro and using it from a function:

void test_MY_TEST_case1(void) { MY_TEST_MACRO(28, 32); }
void test_MY_TEST_case2(void) { MY_TEST_MACRO(27, 33); }
void test_MY_TEST_case3(void) { MY_TEST_MACRO(12, 56); }

It's not much more verbose and it would actually work.

**Comment by ranserme** [comment]
Hello Mark, 
Thank you very much for your prompt answer. 
The point is that I wanted to name the function depending on the given parameters which is a quite long list in my real test case. => Actually it adds quite some verbose :-) 

=> No other method to parse the pre-processed file to generate the test_runner e.g. some option in the *.yml file ?

**Comment by mvandervoord** [comment]
I think if you enable :use_test_preprocessor like so, it might do what you are looking for. If I remember correctly, it causes things to get rebuilt that might not otherwise be necessary, because it has to preprocess the test file each time, which makes it appear to have changes when it doesn't maybe? In any case, you enable it like so:

```
:project:
    :use_test_preprocessor: TRUE
```

**Comment by ranserme** [comment]
Hi, 
I already tried your suggestion, when I read https://github.com/ThrowTheSwitch/Ceedling/issues/241
and unfortunately it doesn't work. It's why I have opened this issue. 
Anyway for the time being I'll do it using some more verbose. 
Thank you very much for your support.

**Comment by Miv13** [comment]
Are there any news on this topic?
